### DISCUSSION TITLE
Is it feasible that the container use webui.sh -f to start A1111's webui?

### DISCUSSION THREAD

**Discussion by Herbert-Zheng** [AbdBarho/stable-diffusion-webui-docker#611] [Ideas]
Thanks for your awesome work first! I think it would be easier to enable or disable extensions if we used `webui.sh` to start A1111's webui instead of executing `webui.py` directly. This is because `webui.sh` acts as an external controller that can restart the webui after changes in extensions. We can add `-f` to force `webui.sh` to execute as root without switching to another user. Will this cause any potential problems?

**Comment by AbdBarho** [comment]
Best thing I can recommend is to try, maybe it works.
you can change this line
https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/master/services/AUTOMATIC1111/Dockerfile#L65

You can also update the docker compose file to make the container restart always
https://docs.docker.com/config/containers/start-containers-automatically/

**Reply by Herbert-Zheng** [reply]
I changed the Dockerfile and it works fine. This (running the script as root) might not be the best practice, but I believe this can be default. Would there be any obstacle that stops you from changing it?
